rustworkx-core/src/connectivity/cycle_basis.rs

```diff
@@ -114,27 +191,67 @@ where
         graph_nodes = graph_nodes.difference(&temp_hashset).copied().collect();
         root_node = None;
     }
-    cycles
+    if want_edges {
+        EdgesOrNodes::Edges(cycles_edges)
+    } else {
+        EdgesOrNodes::Nodes(cycles_nodes)
+    }
+}
+
+/// Enum for custom return types of `cycle_basis()`.
+pub enum EdgesOrNodes<N, E> {
+    Nodes(Vec<Vec<N>>),
+    Edges(Vec<Vec<E>>),
+}
+/// Functions used to unwrap the desired datatype of `EdgesOrNodes`.
+impl<N, E> EdgesOrNodes<N, E> {
+    pub fn unwrap_nodes(self) -> Vec<Vec<N>> {
+        use EdgesOrNodes::*;
+        match self {
+            Nodes(x) => x,
+            Edges(_) => vec![],
+        }
+    }
+    pub fn unwrap_edges(self) -> Vec<Vec<E>> {
+        use EdgesOrNodes::*;
+        match self {
+            Edges(x) => x,
+            Nodes(_) => vec![],
+        }
+    }
 }
```

Comment:
Yes, since we're using a custom return type that can't be interpreted natively as `Vec<Vec<NodeId>>` or `Vec<Vec<EdgeId>>` the values need to be unwrapped. 

Although this could technically be done inside of the public-facing functions that were part of your review, we'd have to rewrite the same procedure twice over (in testing and in the public-facing functions, which is redundant.